### PR TITLE
[DBX-88682] add 562 state logging

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- Add job to perform wholistic state logging
- [Follow up to this work (split for size)](https://github.com/department-of-veterans-affairs/vets-api/pull/17941)

## Related issue(s)

- [Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/88682)
- [High level state repair document](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/untouched_submission_audit/526_state_repair_tdd.md)
- ['paranoid success' document](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/paranoid_success_submissions.md)
- [Scope breakdown document](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/526_scopes.md)

## Testing done

- [x] *New code is covered by unit tests*
- Previously we had no insight into what 'failed' Form526Submissions existed in our database. This logging adds that clarity as the culmination of our 526 'silent failure' code yellow.
- This code has been mostly taken from scripts that were being run in a production command line for the last year. This has served as an organic refining / testing process for the underlying logic. The code itself, as written has *also* been run in a production command line. This was required to confirm it was working and also to check the information against our actual data.

## What areas of the site does it impact?
Form526Submission state. This is not a user facing change, but it provides the final stitches in a saftey net that will prevent future silent 526 failures

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (see above)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (this is the next ticket. we need this live before we can build this)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature (n/a)

### CODE REVIEW

File: app/sidekiq/form526_state_logging_job.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  end
+
+  def load_primary_exhausted
+    subs = timeboxed_submissions
+      .where(submitted_claim_id: nil)
+      .includes(:form526_job_statuses)
+    # this loop-and-lookup is a hack to avoid postgres timeouts
+    ids = subs.select do |sub|
+      sub.form526_job_statuses.any? do |job|
+        job.job_class == 'SubmitForm526AllClaim' && JOB_FAILURE_STATUSES.include?(job.status)
```
[the remark this answers is not in the document]
Comment (SamStuckey) [reply app/sidekiq/form526_state_logging_job.rb:84]: job is the most logically correct name for the variable and is not a reference to the Job that is running

File: app/sidekiq/form526_state_logging_job.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      total_incomplete_type: Form526Submission.incomplete_type.pluck(:id).sort,
+      total_failure_type: Form526Submission.failure_type.pluck(:id).sort
+    }
+  end
+
+  def sub_arel
+    @sub_arel ||= Form526Submission.arel_table
+  end
+
+  def combined_pending_types_for(submissions)
```
[the remark this answers is not in the document]
Comment (SamStuckey) [reply app/sidekiq/form526_state_logging_job.rb:51]: that would be ridiculous

File: app/sidekiq/form526_state_logging_job.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  END_DATE = Date.today.beginning_of_day
+  START_DATE = END_DATE - 1.week
+
+  def initialize(start_date: START_DATE, end_date: END_DATE)
+    @start_date = start_date
+    @end_date = end_date
+  end
+
+  def perform
+    Rails.logger.info('Form 526 State Data', state_log)
```
[the remark this answers is not in the document]
Comment (SamStuckey) [reply app/sidekiq/form526_state_logging_job.rb:20]: this is incorrect, the uses are very different and valid

File: app/sidekiq/form526_state_logging_job.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  def backup_submissions
+    @backup_submissions ||= timeboxed_submissions
+                            .joins(:form526_job_statuses)
+                            .where(form526_job_statuses: { job_class: 'BackupSubmission', status: 'success' })
+  end
+
+  def timeboxed_submissions
+    @timeboxed_submissions ||= Form526Submission
+                               .where(sub_arel[:created_at].gt(start_date))
```
[the remark this answers is not in the document]
Comment (SamStuckey) [reply app/sidekiq/form526_state_logging_job.rb:63]: the alternative would be a helper which seems like overkill

**Review by sethdarragile6 — APPROVED** [review]

**Review by freeheeling — APPROVED** [review]
Whoa, what a job spec!

**Review by ericboehs — APPROVED** [review]

**Pull request merged**